serve/tests/unittest/test_staging_engine.py

```diff
@@ -194,28 +191,29 @@ def test_multiple_requests_preempt():
 
     # Due to asynchronious nature of request submission and processing, we cannot
     # track exactly on each step certain request is processed
-    # but we can catch a pattern that it is two requests processed, then 1 and then again 2
+    # but we can catch a pattern that it is two requests processed simultaneously,
+    # then 1 and it is finished and then the latest and it should be  finished as well
     stage = 0
 
     for s in steps:
         if stage == 0 and len(s.outputs) == 2:
             stage = 1
-        if stage == 1 and len(s.outputs) == 1:
+        elif stage == 1 and len(s.outputs) == 1:
             stage = 2
-        if stage == 2 and len(s.outputs) == 2:
+            assert s.outputs[0].is_finished
+        elif stage == 2 and len(s.outputs) == 1:
             stage = 3
-        if stage == 3 and len(s.outputs) == 1:
-            stage = 4
             assert s.outputs[0].is_finished
-        if stage == 4 and len(s.outputs) > 1:
-            stage = 5
+        elif stage == 3 and len(s.outputs) > 0:
+            stage = 4
 
-    assert stage == 4
+    assert stage == 3
 
 
 # Test to verify if evicted request from active batch which in intermediate
 # state exceeding the max_num_batched_tokens can be processed successfully and will
 # not hang the server in infinite attempt to return it back to the active loop
+@pytest.mark.xfail
```

Comment:
@masahi @yelite Given this PR is mostly about adding more testcases, I'd like to merge this first and fix afterwards. Please follow-up in the next PR.